bin/propolis-server/src/lib/migrate/mod.rs

```diff
@@ -261,13 +261,15 @@ pub async fn source_start(
     ));
     info!(log, "Migration Source");
 
-    let mut context = rqctx.context().context.lock().await;
-    let context =
-        context.as_mut().ok_or_else(|| MigrateError::InstanceNotInitialized)?;
+    let inst_context = tokio::sync::MutexGuard::try_map(
```

Comment:
TIL about `MutexGuard::try_map` :open_mouth:

Reply:
It's super convenient!